migrate/v1/migratev1.go

```diff
@@ -68,28 +96,114 @@ func Migrate(root, driverName string, ls layer.Store, is image.Store, ts tag.Sto
 	return nil
 }
 
-func migrateImages(root string, ls graphIDRegistrar, is image.Store, ms metadata.Store, mappings map[string]image.ID) error {
+// CalculateLayerChecksums walks an old graph directory and calculates checksums
+// for each layer. These checksums are later used for migration.
+func CalculateLayerChecksums(root string, ls checksumCalculator, mappings map[string]image.ID) {
 	graphDir := filepath.Join(root, graphDirName)
-	if _, err := os.Lstat(graphDir); err != nil {
-		if os.IsNotExist(err) {
-			return nil
+	workQueue := make(chan string, runtime.NumCPU()*3)
+
+	wg := sync.WaitGroup{}
+
+	// spawn some extra workers also for maximum performance because the process is bounded by both cpu and io
+	for i := 0; i < runtime.NumCPU()*3; i++ {
```

Comment:
This also seems a little weird.
It will only add to the wait group for `runtime.NumCPU()*3` and that is it, but really shouldn't we be adding for each dir?